File: 백준/Platinum/13482. Java2016/Java2016.cc
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n; cin >> n;
    cout << "a=?max?\n";
    for(char c = 'b'; c <= 'p'; c++) cout << c << "=" << (char)(c-1) << "max" << (char)(c-1) << '\n';
    cout << "q=pmaxpmaxpmaxpmaxp\n";
    cout << "r=q/q\n";
    cout << "s=q-q\n";
    for(char c = 't'; c <= 'z'; c++){
        char pre = (c == 't') ? 'r' : (char)(c-1);
        cout << c << '=' << pre << '+' << pre << '\n';
    }
    if(n){
        vector<char> ans;
        char bit[] = {'r', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
        for(int i = 0; i < 8; i++) if(n & (1<<i)) ans.push_back(bit[i]);
        cout << ans[0];
        for(int i = 1; i < ans.size(); i++) cout << '+' << ans[i];
    }
    else cout << 's';
    return 0;
}

File: 백준/Platinum/13482. Java2016/Java2016.py
import sys
input = lambda: sys.stdin.readline().rstrip()
print = lambda *args, sep=" ", end="\n": sys.stdout.write(sep.join(map(str, args)) + end)

from string import ascii_lowercase

def main():
    n = int(input())
    a_z = ascii_lowercase
    print('a=?max?')
    for i in range(1, 16): print(f'{a_z[i]}={a_z[i-1]}max{a_z[i-1]}')
    print('q=pmaxpmaxpmaxpmaxp')
    print('r=q-q')
    print('s=q/q')
    for i in range(19, 26): print(f'{a_z[i]}={a_z[i-1]}+{a_z[i-1]}')
    if n:
        ans = []
        bit = ['s', 't', 'u', 'v', 'w', 'x', 'y', 'z']
        for i in range(8):
            if n & (1<<i): ans.append(bit[i])
        print('+'.join(ans))
    else: print('r')
    return
if __name__ == "__main__":
    main()